packages/x-data-grid-premium/src/models/dataGridPremiumProps.ts

```diff
@@ -209,4 +234,49 @@ export interface DataGridPremiumPropsWithoutDefaultValue<R extends GridValidRowM
    * @param {GridGetRowsError | GridUpdateRowError} error The data source error object.
    */
   onDataSourceError?: (error: GridGetRowsError<GridGetRowsParams> | GridUpdateRowError) => void;
+  /**
+   * The pivot model of the grid.
+   * Will be used to generate the pivot data.
+   * In case of `pivotActive` being `false`, the pivot model is still used to populate the pivot panel.
+   */
+  pivotModel?: GridPivotModel;
+  /**
+   * Callback fired when the pivot model changes.
+   * @param {GridPivotModel} pivotModel The new pivot model.
+   */
+  onPivotModelChange?: (pivotModel: GridPivotModel) => void;
+  /**
+   * If `true`, the data grid will show data in pivot mode using the `pivotModel`.
+   * @default false
+   */
+  pivotActive?: boolean;
+  /**
+   * Callback fired when the pivot active state changes.
+   * @param {boolean} isPivotActive Whether the data grid is in pivot mode.
+   */
+  onPivotActiveChange?: (isPivotActive: boolean) => void;
+  /**
+   * If `true`, the pivot side panel is visible.
+   * @default false
+   */
+  pivotPanelOpen?: boolean;
```

Comment:
Should this be in the `DataGridPremiumPropsWithDefaultValue` interface instead?

Reply:
I left it here to keep `pivotPanelOpen` and `onPivotPanelOpenChange` together.
It defaults to `undefined`, which is falsy, so it doesn't impact the behavior.
The same applies to `pivotActive` prop.